tests/test_bs.py

```diff
@@ -115,26 +115,45 @@ def query_bootstrap_es(indx, namespace, bootstrap_po_name):
     return None
 
 
-def query_es_client_bootstrap(indx, namespace, client_po_name):
+def query_message(indx, namespace, client_po_name, msg):
+    # TODO : break this to smaller functions ?
     es = get_elastic_search_api()
     fltr = Q("match_phrase", kubernetes__namespace_name=namespace) & \
-           Q("match_phrase", kubernetes__pod_name=client_po_name) & Q("match_phrase", M='discovery_bootstrap')
+           Q("match_phrase", kubernetes__pod_name=client_po_name) & Q("match_phrase", M=msg)
     s = Search(index=indx, using=es).query('bool', filter=[fltr])
     hits = list(s.scan())
 
-    print("Number of Hits in ES: {0}".format(len(hits)))
-    s = set([h.N for h in hits])
-    return len(s)
-
-
-def query_es_gossip_message(indx, namespace, client_po_name):
-    es = get_elastic_search_api()
-    fltr = Q("match_phrase", kubernetes__namespace_name=namespace) & \
-           Q("match_phrase", kubernetes__pod_name=client_po_name) & Q("match_phrase", M='new_gossip_message')
-    s = Search(index=indx, using=es).query('bool', filter=[fltr])
-    hits = list(s.scan())
-
-    print("Number of gossip Hits: {0}".format(len(hits)))
+    print("====================================================================")
+    print("Report for `{0}` in deployment -  {1}  ".format(msg, client_po_name))
+    print("Number of hits: {0}", len(hits))
+    print("====================================================================")
+    print("Benchmark results:")
+    ts = [hit["T"] for hit in hits]
+
+    first = datetime.strptime(min(ts).replace("T", " ",).replace("Z", ""), "%Y-%m-%d %H:%M:%S.%f")
+    last = datetime.strptime(max(ts).replace("T", " ",).replace("Z", ""), "%Y-%m-%d %H:%M:%S.%f")
+
+    delta = last - first
+    print("First: ", first)
+    print("Last: ", last)
+    print("Delta: ", delta)
+    # TODO: compare to previous runs.
+    print("====================================================================")
+    print("Pods that didn't hit:")
```

Comment:
Since this requires additional query, this needs to be done only if it required. Also, not for all the messages we expect to have all the clients in the query output

Reply:
I get the comment about that sometimes it's not required. however currently, we DO expect all clients to show a certain message, I also don't want this function to be too complicated. I think queries with different requirements should get their own method. if they do share some code we can take it out and use it in both.